atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionExpectationsTracker.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl;
+
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.transaction.api.ExpectationsDataSupplier;
+
+public interface TransactionExpectationsTracker extends ExpectationsDataSupplier {
+    void trackBytesRead(TableReference tableRef, long bytesRead);
```

Comment:
agreed, this may be just bad naming on my part

ExpectationsDataSupplier is meant to allow us to query data/stats from the transaction

TransactionExpectationsTracker is meant to allow query & also updates (this is where we track things like - ok transaction read X bytes)

Is the design flawed? Or do I just need better naming? Both perhaps?